Add Wiki section to the documentation (#84)
davenverse/shellfish: docs/wiki/file_manipulation.md

# Handling files and doing operations

> Beyond simple read and write operations, this section allows you to interact directly with the file system. There are functions for creating and deleting files and directories, creating temporary files for short-term use, and managing file and directory permissions for added control, among other useful methods.

Reviewer: ```suggestion
Beyond simple file reading and writing operations, Shellfish allows you to interact directly with the file system. There are functions for creating and deleting files and directories, creating temporary files for short-term use and managing file and directory permissions for added control, among other useful methods.
```

## Creating files and directories

In this section, you will see how to create new files and directories, as well as delete existing ones.

### `createFile`    

> Creates a new file in the specified path, failing if the parent directory does not exist. You can also specify some permissions if you wish. To see what the `exists` function does, see [the reference](#exists):

Reviewer: ```suggestion
Creates a new file in the specified path, failing if the parent directory does not exist. It optionally accepts file permissions. To see what the `exists` function does, see [the reference](#exists):
```

```

@:choice(static)

```scala mdoc:compile-only
import shellfish.FilesOs

val path = Path("path/to/create/file.txt")

FilesOs.createFile(path) >> FilesOs.exists(path) // Should return true
```

@:choice(fs2)

```scala mdoc:compile-only
import fs2.io.file.Files

val path = Path("path/to/create/file.txt")

Files[IO].createFile(path) >> Files[IO].exists(path) // Should return true
```

@:@

### `createDirectories`

Creates all the directories in the path, with the default permissions or with the supplied ones:

@:select(api-style)

@:choice(syntax)

```scala mdoc:compile-only
val directories = Path("here/are/some/dirs")
val path = directories / Path("file.txt")

directories.createDirectories >> path.createFile
```

@:choice(static)

```scala mdoc:compile-only
val directories = Path("here/are/some/dirs")
val path = directories / Path("file.txt")

FilesOs.createDirectories(directories) >> FilesOs.createFile(path)
```

@:choice(fs2)

```scala mdoc:compile-only
val directories = Path("here/are/some/dirs")
val path = directories / Path("file.txt")

Files[IO].createDirectories(directories) >> Files[IO].createFile(path)
```

@:@

### `createTempFile`

> This function creates a temporary file that gets automatically deleted by the operating system. It optionally accepts multiple parameters such as a directory, a prefix (the name of the file), a suffix (the extension of the file) and permissions. It returns the path of the newly created file:

Reviewer: ```suggestion
This function creates a temporary file that gets automatically deleted . It optionally accepts multiple parameters such as a directory, a prefix (to the name of the file), a suffix (like the extension of the file) and permissions. It returns the path of the newly created file:
```

FilesOs.createFile(path) >>
  FilesOs.write(path, "TOP SECRET 🚫, MUST DELETE") >>
    FilesOs.delete(path)
```

@:choice(fs2)

```scala mdoc:compile-only
for
  _ <- Files[IO].createFile(path)

  _ <- Stream.emit("TOP SECRET 🚫, MUST DELETE")
        .through(Files[IO].writeUtf8(path))
        .compile
        .drain

  _ <- Files[IO].delete(path)
yield ()
```

@:@

### `deleteIfExists`

Similar to `delete`, but returns `true` if the deletion was successful instead of failing: 

@:select(api-style)

@:choice(syntax)

```scala mdoc:compile-only
path.deleteIfExists >>= 
  (deleted => IO.println(s"Was the file deleted? $deleted"))
```

@:choice(static)

```scala mdoc:compile-only
FilesOs.deleteIfExists(path) >>= 
  (deleted => IO.println(s"Was the file deleted? $deleted"))
```

@:choice(fs2)

```scala mdoc:compile-only
Files[IO].deleteIfExists(path) >>= 
  (deleted => IO.println(s"Was the file deleted? $deleted"))
```

@:@

### `deleteRecursively`

With the previous functions, the directory had to be empty to be deleted. The difference with this method is that it recursively deletes all files or folders contained inside it, optionally following symbolic links if specified: 

> Recursively deletes all files or folders, following symbolic links if specified. 

Reviewer: Doesn't this line repeat the previous one?

Author: You are right >:)

Note that, unlike the previous functions, this one will not fail if the directories are empty or do not exist:

@:select(api-style)

@:choice(syntax)

```scala 3
val dirs = Path("this/folders/will/be/created/and/deleted")

for 
  _ <- dirs.createDirectories
  _ <- dirs.deleteRecursively // Will delete all of them!
yield ()
```

@:choice(static)

```scala 3
val dirs = Path("this/folders/will/be/created/and/deleted")

for 
  _ <- FilesOs.createDirectories(dirs)
  _ <- FilesOs.deleteRecursively(dirs) // Will delete all of them!
yield ()
```

@:choice(fs2)

```scala 3

val dirs = Path("this/folders/will/be/created/and/deleted")

for 
  _ <- Files[IO].createDirectories(dirs)
  _ <- Files[IO].deleteRecursively(dirs) // Will delete all of them!
yield ()
```

@:@

## File operations

> Working directly with files is a common need in many scripting scenarios. This library provides essential functions for renaming, moving, and copying files, allowing you to efficiently manage and organise your data. 

Reviewer 2: ```suggestion
Shellfish provides essential functions for renaming, moving, and copying files, allowing you to efficiently manage your data. These are especially useful in scripting scenarios.
```

### `copy`

> Copies a file from a source path to a target path. The method will fail if the destination path already exists; to avoid this behaviour, you can pass flags to e.g., replace the contents in the destination:

Reviewer: ```suggestion
Copies a file from a source path to a target path. The method will fail if the destination path already exists; to avoid this behaviour, you can, for example, pass flags to replace the contents at destination:
```
